src/core/pyproxy.ts

```diff
@@ -2304,10 +2332,15 @@ export class PyAwaitableMethods {
    * @private
    */
   _ensure_future(): Promise<any> {
-    if (this.$$.promise) {
-      return this.$$.promise;
+    const { shared } = _getAttrsQuiet(this);
+    if (shared.promise) {
+      return shared.promise;
+    }
+    const ptr = shared.ptr;
+    if (!ptr) {
+      // Destroyed and promise wasn't resolved. Raise error!
+      _getAttrs(this);
```

Comment:
Here we've already checked that the proxy has been destroyed, so we know it will raise an error. We're just calling it so that it will raise the normal error message rather than having to duplicate the error raising logic here (or factor it out which would perhaps make more sense).